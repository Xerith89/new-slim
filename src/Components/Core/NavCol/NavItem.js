import React from 'react'
import PropTypes from 'prop-types'

export default function NavItem(props) {
    return (
        <div>
            {props.links.map((element, i) => {   
                return ( 
                <React.Fragment key={i}>
                    <a href="/" className=" btn btn-block btn-secondary m-0" >{element}</a>
                    <hr style={{margin: '0rem', borderColor: '#2F4F4F'}}/>
                </React.Fragment>)
            })}
        </div>)
}

NavItem.propTypes = {
    links: PropTypes.array
};