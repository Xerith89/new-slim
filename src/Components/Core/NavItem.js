import React from 'react'

export default function NavItem(props) {
    return (
        <div>
             <ul className="list-group list-group-flush">
                <h3 className="navbar-dark bg-dark text-white list-group-item">{props.title}</h3>
                {props.links.map((element, i) => {
                    return <li className="list-group-item" key={i}>{element}</li>
                })}
            </ul>
        </div>
    )
}
