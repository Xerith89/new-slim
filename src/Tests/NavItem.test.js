import React from 'react';
import NavItem from '../Components/Core/NavCol/NavItem';
import {render, cleanup} from '@testing-library/react';


afterEach(cleanup)

test('NavItem component renders to the DOM', () => {
    const {getByText} = render(<NavItem links={["navItemTest"]} />);
    expect(getByText('navItemTest')).toMatchSnapshot();
});