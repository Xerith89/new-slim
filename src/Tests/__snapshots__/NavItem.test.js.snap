// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem component renders to the DOM 1`] = `
<a
  class=" btn btn-block btn-secondary m-0"
  href="/"
>
  navItemTest
</a>
`;
